### PR TITLE
Feature/singlestore type tests

### PR DESCRIPTION
**Pull request by Rodriguespn**
The type tests test the type safety of the drizzle-orm operations and are located in the “drizzle-orm/type-tests” directory.

Copied and renamed every MySQL reference to singlestore on the `drizzle-orm/src/type-tests` directory.

Also, I removed the foreign key and check constraint type tests, since they are not supported by singlestore.

Note: Defined `intersectAll` and `excludeAll` sql functions for SingleStore, since they were defined for MySQL and not for SingleStore and were used on the type tests

### CODE REVIEW

File: drizzle-orm/src/singlestore-core/query-builders/select.ts [hunk 1]
```diff
@@ -1028,6 +1114,48 @@ export const unionAll = createSetOperator('union', true);
...
+ * .intersectAll(
+ *   db.select({
+ *     productId: vipCustomerOrders.productId,
+ *     quantityOrdered: vipCustomerOrders.quantityOrdered
+ *   })
+ *   .from(vipCustomerOrders)
+ * );
+ * ```
+ */
+export const intersectAll = createSetOperator('intersect', true);
```
Comment (mitchwadair) [inline drizzle-orm/src/singlestore-core/query-builders/select.ts:1157]: sorry if this is a dumb question, maybe I'm missing context here.. Are there supposed to be two declarations of each `intersectAll` and `exceptAll`?

**Review by mitchwadair — APPROVED** [review]
Just to be sure, this is just the copy/paste/rename of the type tests from MySQL? Is there anything more specific to review?

File: drizzle-orm/src/singlestore-core/query-builders/select.ts (hunk 1 above)
Comment (Rodriguespn) [reply drizzle-orm/src/singlestore-core/query-builders/select.ts:1157]: I also found it very odd that there were two different declarations of `intersectAll` and `exceptAll`, even stranger that one is exported and the other is not but this was how other functions were implemented in MySQL such as `intersect` or `union` so I copied it.

**Comment by Rodriguespn** [comment]
@mitchwadair yes, it's just the copy/paste/rename of the type tests from MySQL. I also removed the foreign key type tests from MySQL since SingleStore doesn't support FKs

**Pull request merged**